src/connection-manager/wsHandling.ts

```diff
@@ -225,3 +235,69 @@ export async function handleWsMessageLedgerEnableAmendments(
     }),
   )
 }
+
+/**
+ * Backtracking a network to update amendment status in case of Websocket disconnection.
+ *
+ * @param network - The network being tracked.
+ * @param url - The Faucet URL of the network.
+ * @returns Void.
+ */
+async function backtrackNetworkAmendmentStatus(
+  network: string,
+  url: string,
+): Promise<void> {
+  try {
+    log.info(`Backtracking to update amendment status for ${network}...`)
+    const client = new Client(url)
+    await client.connect()
+    const ledgerResponse: LedgerResponse = await client.request({
+      command: 'ledger',
+      ledger_index: 'validated',
+    })
+    const currentLedger = ledgerResponse.result.ledger_index
+
+    // a flag + 1 ledger typically comes in every 10 to 15 minutes.
+    const fourFlagPlusOneLedgerBefore =
+      (Math.floor(currentLedger / 256) - 3) * 256 + 1
+
+    for (
+      let index = fourFlagPlusOneLedgerBefore;
+      index < currentLedger;
+      index += 256
+    ) {
+      const ledger: LedgerResponse = await client.request({
+        command: 'ledger',
+        transactions: true,
+        ledger_index: index,
+        expand: true,
+      })
+
+      await handleWsMessageLedgerEnableAmendments(
+        ledger as LedgerResponseCorrected,
+        network,
+      )
+    }
+
+    await client.disconnect()
+
+    log.info(`Finished backtracked amendment status for ${network}...`)
+  } catch (error) {
+    log.error(
+      `Failed to backtrack amendment status for ${network} due to error: ${String(
```

Comment:
It shouldn’t be necessary to wrap in String since toString is automatically called anyways

Reply:
This is to avoid linting issue